### PR TITLE
feat: drawer component

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Introduces a new drawer component that can be used on it's own for smaller screen size experiences. It will also be used for a new dialog component in #2853.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Relates to #2853

## Mobile & Desktop Screenshots/Recordings

CleanShot 2024-03-20 at 22 23 44

You can swipe the drawer away, click close to close it, or click or tap outside the drawer to close it.

https://github.com/open-sauced/app/assets/833231/e622ba46-7eca-4ae6-9be7-f222604d2d4f

## Steps to QA

1. Visit https://deploy-preview-2992--design-insights.netlify.app/?path=/docs/components-shared-drawer--docs to take the component for a test drive.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [x] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/primitives/drawer-primitives.tsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import clsx from "clsx";
```
Comment (nickytonline) [inline components/primitives/drawer-primitives.tsx:1]: ShadCN doing it's thing.

File: components/shared/Drawer.tsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  description: React.ReactNode;
+  children: React.ReactNode;
+  trigger: React.ReactNode;
+  showCloseButton?: boolean;
+}
+
+export const Drawer = ({ title, description, children, showCloseButton = true, trigger }: DrawerProps) => {
+  return (
+    <InternalDrawer>
+      <DrawerTrigger>{trigger ?? <Button variant="primary">Open</Button>}</DrawerTrigger>
```
Comment (nickytonline) [inline components/shared/Drawer.tsx:24]: The default trigger might not be useful. Probably better to always require one.

File: styles/globals.css [hunk 3]
```diff
@@ -220,3 +220,9 @@ svg:has(> #Mail) {
 .workspace-drop-down > [role="combobox"] {
   max-width: 14rem;
 }
+
+/* TODO: Sort out configuring ShadCN as we've only ever brought in components from it,
+but never configured it. */
+.bg-muted {
```
Comment (nickytonline) [inline styles/globals.css:226]: This is for the universal bar styling @isabensusan. The rest of the styling is already there.

**Review by isabensusan — APPROVED** [review]
This is one good looking drawer!!! tysm

scared-comedy-central-gif-by-chescaleigh

**Review by jpmcb — APPROVED** [review]
Looks solid! 👏🏼

**Pull request merged**